src/components/MoneyReportHeader.tsx

```diff
@@ -224,7 +226,7 @@ function MoneyReportHeader({
                 shouldShowBackButton={shouldUseNarrowLayout}
                 onBackButtonPress={onBackButtonPress}
                 // Shows border if no buttons or next steps are showing below the header
-                shouldShowBorderBottom={!(shouldShowAnyButton && shouldUseNarrowLayout) && !(shouldShowNextStep && !shouldUseNarrowLayout) && !allHavePendingRTERViolation}
+                shouldShowBorderBottom={!isMoreContentShown && !allHavePendingRTERViolation}
```

Comment:
@bernhardoj hmm.. Looks like I am missing something here. Can you please help me understand why we need to replace the existing conditions with `isMoreContentShown`? Both seem to be different.

Reply:
I'm trying to simplify it. We need to set `shouldShowBorderBottom` to false if `hasScanningReceipt` is true, but instead of appending it to the existing condition (&& !hasScanningReceipt), I just replaced it with `isMoreContentShown`.

Without simplifying:
```
!(shouldShowAnyButton && shouldUseNarrowLayout) && !(shouldShowNextStep && !shouldUseNarrowLayout) && !hasScanningReceipt && !allHavePendingRTERViolation
```

`isMoreContentShown` condition
```
shouldShowNextStep || hasScanningReceipt || (shouldShowAnyButton && shouldUseNarrowLayout)
```

If we reverse this (!isMoreContentShown), it becomes,
```
!shouldShowNextStep && !hasScanningReceipt && !(shouldShowAnyButton && shouldUseNarrowLayout)
```

repositioned:
```
!(shouldShowAnyButton && shouldUseNarrowLayout) && !shouldShowNextStep && !hasScanningReceipt
```

You can see that `hasScanningReceipt` and `shouldShowAnyButton` condition is the same. The only difference is `shouldShowNextStep`.

Next step component is put outside of the HeaderWithBackButton, no matter whether it's `shouldUseNarrowLayout` or not which has its own border bottom.
https://github.com/Expensify/App/blob/8375abea3a377e7416404addde255b6ac8530a9b/src/components/MoneyReportHeader.tsx#L335-L339
https://github.com/Expensify/App/blob/8375abea3a377e7416404addde255b6ac8530a9b/src/components/MoneyReportHeader.tsx#L303

So, we don't want the extra border bottom from the HeaderWithBackButton anymore.